azavea/raster-foundry
# Cache Refactor in Tile Server

## Overview

This PR makes a few big changes:
 - Removes scaffeine and heap backed memcached client inside the tile server
 - Removes TTLs on tiles
 - Makes request for scene data only when the web tile intersects the extent of the web tile
 - Introduces separate execution contexts for cache and attribute store interactions + separates out blocking operations inside the tile server with a blocking execution context
 - Adds settings for various knobs via environment variables for caches
 - Various cleanup on formatting

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

See https://github.com/azavea/raster-foundry/issues/2306#issuecomment-317564849 for various benchmarks

### Notes

Opened https://github.com/azavea/raster-foundry/issues/2304 to deal with new Hikari issue. Tile server should like be refactored to work in similar fashion to the API server for consistency.

## Testing Instructions

 * Start servers
 * Verify that things get cached -- there are still initial timeouts on thumbnails; however, that likely requires additional work. I think we are just making too big of a request for tiles that we are resampling anyway.


Closes #2306
